File: lib/meeting_details.dart
class MeetingDetails {
  String meetingId;
  String meetingPin;
  String displayName;
  bool isInitialAudioOn;
  bool isInitialVideoOn;
  String? hostToken;

  MeetingDetails(
      {required this.meetingId,
      required this.meetingPin,
      required this.displayName,
      this.isInitialAudioOn = false,
      this.isInitialVideoOn = false,
      this.hostToken,
      
      });

  Map<String, dynamic> toJson() {
    return {
      'meetingId': meetingId,
      'meetingPin': meetingPin,
      'displayName': displayName,
      'isInitialAudioOn': isInitialAudioOn,
      'isInitialVideoOn': isInitialVideoOn,
      'hostToken': hostToken,
    };
  }
}

File: android/src/main/kotlin/jio/coresdk/coresdk_plugin/MeetingDetails.kt
package jio.coresdk.coresdk_plugin
import com.google.gson.Gson

data class MeetingDetails(
    var meetingId:String,
    var meetingPin:String,
    var displayName:String,
    var isInitialAudioOn:Boolean,
    var isInitialVideoOn:Boolean,
    var hostToken: String?
) {
    fun toJsonString() : String {
        return Gson().toJson(this, MeetingDetails::class.java)
    }

    companion object {
        fun fromJson(json : String) : MeetingDetails? {
            return Gson().fromJson(json, MeetingDetails::class.java)
        }
    }
}
